src/storage/service.ts

```diff
@@ -924,4 +925,24 @@ export class StorageService {
   async download (commp: string | CommP, options?: DownloadOptions): Promise<Uint8Array> {
     return await this.providerDownload(commp, options)
   }
+
+  /**
+   * Get proof set roots for a given proof set ID
+   * @param proofSetId - The ID of the proof set to fetch roots for
+   * @returns Array of root CIDs as CommP objects
+   */
+  async getProofSetRoots (proofSetId: string): Promise<CommP[]> {
```

Comment:
This shouldn't take an argument, we have `this._proofSetId`. A `StorageService` is tied to a proofset. If we take an arbitrary one but we're only querying the one server then we may not be able to get the info about it. A proper "getProofSetRoots" taking a param might have to look up the chain to see the PDP server to query.